scripts/deploy-multidev.sh

```diff
@@ -1,27 +1,81 @@
-export SITE=72e163bd-0054-4332-8bf8-219c50b78581
-# Create local directory and file to log rsync output (local path: "$HOME/sites/docs-backups")
-mkdir ../docs-backups
-mkdir ../docs-backups/`date +%F-%I%p`
-echo "rsync log - deploy to docsfiles Multidev environment on `date +%F-%I%p`" > ../docs-backups/`date +%F-%I%p`/rsync-`date +%F-%I%p`.log
-# Build prod
-rm -rf output_prod/docs
-sculpin generate --env=prod
-cd output_prod
-
-### Get path for deploy from user and deploy only specified doc
-read -p "What doc or directory? (e.g. docs/articles/users/index.html or docs/articles/users/): " path
-[ -z "${path}" ] && path='docs'
-if [ ! -e $path ]
+# Deploy branch to multidev on commit
+# Note: This script uses CircleCI environment variables https://circleci.com/docs/environment-variables
+# Note: PRs from forks not yet supported, see: https://circleci.com/docs/fork-pr-builds
+
+# Deploy any branch except master, dev, test, or live
+if [ "$CIRCLE_BRANCH" != "master" ] && [ "$CIRCLE_BRANCH" != "dev" ] && [ "$CIRCLE_BRANCH" != "test" ] && [ "$CIRCLE_BRANCH" != "live" ]; then
+
+    # Normalize branch name to adhere with Multidev requirements
+    export normalize_branch="$CIRCLE_BRANCH"
+    export valid="^[-0-9a-z]" # allows digits 0-9, lower case a-z, and -
+    if [[ $normalize_branch =~ $valid ]]; then
+        export normalize_branch="${normalize_branch:0:11}"
+        echo "Success: "$normalize_branch" is a valid branch name."
+    else
+        echo "Error: Multidev cannot be created due to invalid branch name: $normalize_branch"
+        exit 1
+    fi
+
+    # Authenticate Terminus
+    ~/documentation/bin/terminus auth login $PANTHEON_EMAIL --password=$PANTHEON_PASS
+
+
+    # Write existing environments for the static docs site to a text file
+    ~/documentation/bin/terminus site environments --site=static-docs > ./env_list.txt
+
+
+    # Filter env_list.txt into a single column for easier verification
+    echo "Existing environments:"
+    tail -n +2 env_list.txt | cut -f1 | tee ./filtered_env_list.txt
+
+
+    # Check env_list.txt, create environment if one does not already exist
+    if grep -Fxq "$normalize_branch" ./filtered_env_list.txt; then
+        echo "Existing environment found for $normalize_branch: http://"$normalize_branch"-static-docs.pantheon.io"
+    else
+        ~/documentation/bin/terminus site create-env --site=static-docs --from-env=dev --to-env=$normalize_branch
+        echo "Multidev created for $normalize_branch: http://"$normalize_branch"-static-docs.pantheon.io"
+        sleep 90 # Wait for multidev to be created
+    fi
+
+    # Update redirect script for the Multidev environment
+    export avoid_redirect="window.location.hostname == '$normalize_branch-static-docs.pantheon.io' ||"
+    sed -i '9i\'"      ${avoid_redirect}"'\' source/_views/default.html
+
+
+    # Regenerate sculpin to reflect new redirect logic
+    bin/sculpin generate --env=prod
+
+
+    # Create log dir
+    mkdir ../docs-backups
```

Comment:
@bmackinney - dir contains backup files and log file, so as it stands the name is accurate. 

But, now that we're talking about backups - we should not be writing backups to Valhalla on the Live environment. I would say it's okay for Multidev because the environments are deleted when merged into master, but on Live we'll have to either automate deletion or manually delete subdirs to avoid issues with highly populated directories.

I'm in favor of removing the rsync backup generation all together. Reverting a deployment should be done using GitHub's revert feature or on the command line with `git revert`. We can upload the rsync log to Valhalla for rsync debugging. Thoughts?